### DISCUSSION TITLE
Building JSDoc through GitHub Actions

### DISCUSSION THREAD

**Discussion by haykam821** [community/community#26213] [Actions]
<p>Hi! Unfortunately, I am pretty lost when it comes to GitHub Actions. I would like to build JSDoc when my repo is pushed to. Is this possible with GitHub Actions, or should I be looking for an alternative?</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Yes, GitHub Actions essentially allows you to do anything you can do in a Docker container, which is nearly limitless. Since GitHub Actions are time-limited though, you would need to build the Action to build your JSDoc and then upload it someplace else to be downloaded or hosted before the Action completes.</p>
<p>I hope that helps! Let us know if you have any other questions.</p>

**Comment by yafp** [comment]
<p>So … there is no way to run for example</p>
<pre><code>npm run jsdoc-generation-script
</code></pre>
<p>to start the jsdoc generation and add them to the current repo?</p>
<p>You can only imagine to handle that with a second target (like github pages), right?</p>

**Comment by lee-dohm** [comment]
<p>You can do what you’re describing, yes. You could create a workflow to monitor for pushes to your default branch, build the documentation from that change, then push those new docs back to the default branch, if you chose. Actions specifically prevents workflows from triggering other workflows, so this is safe from an infinite worker loop.</p>
<p>Personally, I use <a href="https://github.com/lee-dohm/atom-style-tweaks/blob/master/.github/workflows/publish-docs.yml" rel="nofollow noopener">a workflow</a> for my projects to push generated documentation to GitHub Pages because I find having the docs hosted on the web more useful than only in the repository. But it doesn’t <strong>have</strong> to be done that way. As I said, Actions is pretty wide open on what you can do with it.</p>
